### PR TITLE
Security rules with service group and address groups

### PR DESCRIPTION
**Pull request by siddharth-kulshrestha**

### CODE REVIEW

**Comment by siddharth-nutanix** [comment]
/ok-to-test
>Acceptance test run status: success
>Line code coverage is 60.1

File: nutanix/resource_nutanix_network_security_rule.go [hunk 1]
```diff
@@ -1353,6 +1385,26 @@ func expandNetworkRule(prefix string, d *schema.ResourceData) *v3.NetworkSecurit
 	return appRule
 }
 
+func expandReferencePtr(reference map[string]interface{}) *v3.Reference {
```
Comment (bhati-pradeep) [inline nutanix/resource_nutanix_network_security_rule.go]: Is this same as expandReference() helper ?

File: nutanix/resource_nutanix_network_security_rule.go [hunk 2]
```diff
@@ -1414,6 +1466,20 @@ func filterParamsHash(v interface{}) int {
 	return utils.HashcodeString(params["name"].(string))
 }
 
+func flattenReferencesList(refs []*v3.Reference) []map[string]interface{} {
```
Comment (bhati-pradeep) [inline nutanix/resource_nutanix_network_security_rule.go]: can we use flattenArrayReferenceValues() helper method ?

**Review by bhati-pradeep — CHANGES_REQUESTED** [review]

**Comment by siddharth-nutanix** [comment]
/ok-to-test
>Acceptance test run status: failure
>

**Review by bhati-pradeep — APPROVED** [review]

**Comment by siddharth-nutanix** [comment]
/ok-to-test
>Acceptance test run status: success
>Line code coverage is 60.3

**Pull request merged**